extension/js/background.js

```diff
@@ -176,62 +224,87 @@ function handleMessage(data, sender, sendRespones) {
         url: data.url,
     };
     //console.log(item)
-    db.put(item, function callback(err, result){
-        if(!err) {
+    db.put(item, function callback(err, result) {
+        if (!err) {
             console.log('Successfully stored the page: ' + data.url)
-        }
-        else {
+        } else {
             console.log('Error on URL: ' + data.url)
         }
 
     });
-}*/
+}
 
-/*function show_url(){
+function show_url() {
     db.allDocs({
-       include_docs: true, descending:true 
-    }, function(err,doc){
+        include_docs: true,
+        descending: true
+    }, function (err, doc) {
         console.log(doc);
     });
-}*/
+}
 
-/*function search_pouch(query, text, cb, suggestCb ){
-    //console.log(query)
+function search_pouch(query, text, cb, suggestCb) {
+    console.log(query);
+    // ,
+        // filter: function (doc) {
+        //     console.log("Inside Filter");
+        //     // console.log(query);
+        //     if (query.before != false && doc.LastVisitTime <= query.before.getTime() && query.LastVisitTime >= doc.after.getTime()) {
+        //         console.log("Comparison 2");
+        //         return doc;
+        //     }
+        //     else if (doc.LastVisitTime >= query.after.getTime()) {
+        //         console.log("Comparison 1");
+        //         return doc;
+        //     }
+        // }
     db.search({
         query: query.text,
         fields: ['text'],
         include_docs: true
-    }).then(function(res){
-        console.log("RESULTS: ")
-        console.log(res.rows)
-
-        //console.log(doc.LastVisitTime)
-        return makeSuggestions(query, res.rows, cb, suggestCb)
-    }).catch(function(err){
+    }).then(function (res) {
+        // console.log("RESULTS: ")
+        // console.log(res.rows)
+        var results = [];
+        for(var i = 0; i < res.rows.length ; i++) {
+            // console.log(res.rows[i]);
+            var doc = res.rows[i];
+            if (query.before != false) {
+                if (doc.doc.LastVisitTime <= query.before.getTime() && doc.doc.LastVisitTime >= query.after.getTime())
+                    results.push(doc);
```

Comment:
`doc.doc` has the properties for validating the object, but the existing architecture for showing suggestions requires the `doc` object itself. `doc` is the raw object obtained from the DB having the `_id` etc.